parsers/KPX.py

```diff
@@ -164,29 +123,62 @@ def fetch_price(
                 table_prices.iloc[row_idx, col_idx] * 1000
             )  # Convert from Won/kWh to Won/MWh
 
-            data = {
-                "zoneKey": zone_key,
-                "datetime": arw_day.datetime,
-                "currency": "KRW",
-                "price": price_value,
-                "source": "new.kpx.or.kr",
-            }
+            price_list.append(
+                zoneKey=zone_key,
+                datetime=arw_day.datetime,
+                source=KR_SOURCE,
+                price=price_value,
+                currency=KR_CURRENCY,
+            )
+
+    return price_list.to_list()
 
-            all_data.append(data)
 
+def _parse_long_term_prod_data(html: str) -> Dict[datetime, ProductionMix]:
```

Comment:
```suggestion
def _parse_long_term_prod_data(html: str, zone_key: ZoneKey) -> ProductionBreakdownList:
```
You could directly create the list here instead of creating an adhoc dict with timestamps as indexes.

Reply:
I did this on purpose to avoid having to propagate the zone_key and logger arguments where it's not really needed, but I'm open for discussion here

Reply:
I don't really see it as an issue to propagate the zone key and the logger, correctly encapsulating the function's behavior seems like the best way to do it.
Basically by using a custom wraping (datetime index dict of Mixes), the function making use of the parsed results needs some inside knowledge about how things are done in the parsing function. Mainly that the dict is indexed by datetimes and those refer to the time of the production/storage. It's generally not the best practice because ideally you should directly have all you need from the function's signature.

Reply:
Besides, in this case you end up looping twice over the same data, not a big deal because the dataset is small but we could save ourselves one loop.